### PR TITLE
Update url for the Windows package; manually migrate to CUDA 11.1/11.2

### PR DESCRIPTION
**Pull request by leofang**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by leofang** [comment]
@conda-forge-admin, please rerender

**Comment by leofang** [comment]
@jakirkham Any advice for how to make this package support CUDA 11.x without generating extra packages? I think the binary works for all CUDA 11.x despite we built it in the 11.0 docker.

**Comment by jakirkham** [comment]
Sorry not following. Meaning avoiding uploading more Linux packages? As long as the package hash hasn't changed (and the build number hasn't changed), there shouldn't be any new uploads. Should be possible to see on CI what the package filename (including hash) is at the end of the build

**Comment by leofang** [comment]
> Meaning avoiding uploading more Linux packages?

@jakirkham, I meant here I produce 3 packages for 10.1, 10.2, and 11.0, but the last one should also work on any 11.x. Is there a `run_export` thing that I need to fix in order to allow this?

**Comment by leofang** [comment]
> I meant here I produce 3 packages for 10.1, 10.2, and 11.0, but the last one should also work on any 11.x. Is there a `run_export` thing that I need to fix in order to allow this?

Perhaps this is clearer. I want to fix this conflict in this PR to achieve a "manual migration":
```
$ conda create -c conda-forge -n test cudatoolkit=11.1 cutensor
Collecting package metadata (current_repodata.json): done
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: / 
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed                                                                                                                                                                 

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versions

Package cudatoolkit conflicts for:
cudatoolkit=11.1
cutensor -> cudatoolkit[version='10.1|10.1.*|11.0|11.0.*|10.2|10.2.*']

Package __glibc conflicts for:
cutensor -> __glibc[version='>=2.17|>=2.17,<3.0.a0']
@/linux-64::__glibc==2.31=0
cudatoolkit=11.1 -> __glibc[version='>=2.17,<3.0.a0']The following specifications were found to be incompatible with your CUDA driver:

  - feature:/linux-64::__cuda==11.2=0
  - feature:|@/linux-64::__cuda==11.2=0

Your installed CUDA driver is: 11.2
```

**Comment by kkraus14** [comment]
@leofang the recent commit you pushed is what I'd expect to allow the CUDA 11 package to run on any CUDA 11.x version. Let us know if it doesn't work!

File: recipe/meta.yaml [hunk 1]
```diff
@@ -26,18 +26,18 @@ build:
...
+    - copy include\\cutensor.h %LIBRARY_INC%\\                             # [win64]
+    - mkdir %LIBRARY_INC%\\cutensor                                        # [win64]
+    - copy include\\cutensor\\types.h %LIBRARY_INC%\\cutensor              # [win64]
+    - copy lib\\{{ cuda_compiler_version }}\\cutensor.dll %LIBRARY_BIN%\\  # [win64 and cuda_compiler_version in ("10.1", "10.2")]
+    - copy lib\\{{ cuda_compiler_version }}\\cutensor.lib %LIBRARY_LIB%\\  # [win64 and cuda_compiler_version in ("10.1", "10.2")]
+    - copy lib\\11\\cutensor.dll %LIBRARY_BIN%\\                           # [win64 and cuda_compiler_version in ("11.0", )]
+    - copy lib\\11\\cutensor.lib %LIBRARY_LIB%\\                           # [win64 and cuda_compiler_version in ("11.0", )]
   ignore_run_exports_from:
-    - {{ compiler('c') }}    # [linux]
+    - {{ compiler('c') }}     # [linux]
```
Comment (jakirkham) [inline recipe/meta.yaml:37]:
```suggestion
    - {{ compiler('c') }}     # [linux]
    - {{ compiler('cuda') }}  # [cuda_compiler_version == "11.0"]
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -9,15 +9,15 @@ source:
...
   sha256: 954ee22b80d6b82fd4decd42b7faead86af7c3817653b458620a66174e5b89b6  # [linux64]
   sha256: d914a721b8a6bbfbf4f2bdea3bb51775e5df39abc383d415b3b06bbde2a47e6e  # [ppc64le]
   sha256: 031b75946d2ce22f8c2736f7b4216b9377406afcb35012093a66e2801bfe2990  # [aarch64]
-  sha256: 88d3e2c662e601214ef5caa26ca2db0492eab6ec3bf9dc9a746bb1a7b8c3aab2  # [win64]
+  sha256: 8f960b22594a91dbad7e78cc91acb92a2a87607b2f46085fe6ecd6173747467a  # [win64]
 
 build:
-  number: 2
+  number: 3
   skip: True  # [win32 or cuda_compiler_version in ("None", "9.2", "10.0")]
```
Comment (jakirkham) [inline recipe/meta.yaml:21]:
```suggestion
  skip: True  # [win32 or cuda_compiler_version in ("None", "9.2", "10.0", "11.1", "11.2")]
```

**Review by jakirkham — COMMENTED** [review]
Would also suggest the following. `skip` is more preventive at this point since this isn't migrated yet

File: recipe/meta.yaml (hunk 1 above)
Comment (leofang) [reply recipe/meta.yaml:37]: @jakirkham looks like this would make the compilation test fail...

Comment (leofang) [reply recipe/meta.yaml:37]: ...because `cudatoolkit` was not installed at the test stage. But, oddly, the same test passed on Windows!

Comment (leofang) [reply recipe/meta.yaml:37]: 885d4d1 seems to fix it.

Comment (jakirkham) [reply recipe/meta.yaml:37]: Weird would have thought `pin_compatible` in `run` would have been sufficient

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render

File: recipe/meta.yaml [hunk 3]
```diff
@@ -64,6 +65,9 @@ test:
     - {{ compiler('cxx') }}
     - {{ compiler('cuda') }}
```
Comment (jakirkham) [inline recipe/meta.yaml:66]: Ah we might also want to update this for 11

Comment (leofang) [reply recipe/meta.yaml:66]: But we actually need nvcc to test compiling, so I am confused by why keeping nvcc here is not accompanied by a correct cudatoolkit...

File: recipe/meta.yaml [hunk 4]
```diff
@@ -64,6 +65,9 @@ test:
...
     - sysroot_linux-64 2.17  # [linux64]
+    # At this point we are still inside the 11.0 docker, so needs to avoid pulling
+    # down a newer runtime, otherwise the linker would complain
+    - cudatoolkit 11.0  # [linux and cuda_compiler_version == "11.0"]
```
Comment (leofang) [inline recipe/meta.yaml:70]: I need to investigate why we have to do this on Linux but not on Windows.

Comment (leofang) [reply recipe/meta.yaml:70]: I noticed the effect of adding `nvcc` to `ignore_run_exports_from` is that at build time, `cudatoolkit` is no longer installed. Although it's not really used at build time, I don't think this behavior is correct.

File: recipe/run_test.sh [hunk 5]
```diff
@@ -2,6 +2,11 @@
 set -ex
 
 
+if [ $cuda_compiler_version == '11.0' ]; then
+    conda create -n test cudatoolkit=11.2 cutensor
```
Comment (jakirkham) [inline recipe/run_test.sh:6]: Could we list these in `test/requires` instead?

Comment (leofang) [reply recipe/run_test.sh:6]: It's only for debugging for now. Yeah we could move it there once we confirm things are working.

Comment (leofang) [reply recipe/run_test.sh:6]: I don't think this is a valid test, is it? Does the solver fetch `cutensor` from local channel or conda-forge?

**Comment by leofang** [comment]
@conda-forge-admin, please rerender

**Comment by leofang** [comment]
```
Package cudatoolkit conflicts for:
cutensor -> cudatoolkit[version='10.1|10.1.*|11.0|11.0.*|10.2|10.2.*']
cudatoolkit=11.2
```

**Comment by leofang** [comment]
@conda-forge-admin, please rerender

File: recipe/run_test.sh (hunk 5 above)
Comment (jakirkham) [reply recipe/run_test.sh]: The solver treats packages that are built locally as coming from a special channel that has higher priority than any other channel

Comment (leofang) [reply recipe/run_test.sh]: I see, good to confirm.

**Comment by leofang** [comment]
Alright. My hands are tied. Looks like `pin_compatible` isn't working as expected, and the dependency on 11.0 is just not relaxed to 11.x. 

I will leave the PR open for future investigation, as the url change itself isn't critical and I want this PR to also do the migration for us.

**Comment by jakirkham** [comment]
Let’s see if we can get some fresh eyes on this

cc @conda-forge/core

File: recipe/meta.yaml [hunk 6]
```diff
@@ -47,12 +47,14 @@ requirements:
     - {{ compiler('cuda') }}
     - sysroot_linux-64 2.17  # [linux64]
     - libgomp  # [linux]
-    - 7zip  # [win]
+  host:
```
Comment (leofang) [inline recipe/meta.yaml:50]: @isuruf Why do we need an empty host section?

Comment (jakirkham) [reply recipe/meta.yaml:50]: Historically conda-build did not separate `build` in `host` (in fact there was only `build` originally). Listing a separate empty `host` ensures conda-build does split them (though usually it is not needed)

File: recipe/meta.yaml [hunk 7]
```diff
@@ -33,24 +33,28 @@ build:
...
     - '*/libgomp.*'  # [linux]
 
 requirements:
   build:
+    - {{ compiler('c') }}
+    - {{ compiler('cuda') }}
     - sysroot_linux-64 2.17  # [linux64]
     - libgomp  # [linux]
-    - cudatoolkit {{ cuda_compiler_version }}
+  host:
```
Comment (jakirkham) [inline recipe/meta.yaml]: However there was another issue, which is `cudatoolkit` was included in `build`. Really `build` should only be build tools used for their executables (`make`, compilers, etc.). So this likely was causing us issues

Comment (leofang) [reply recipe/meta.yaml]: I am sure I have tried Isuru's change earlier (apart from adding an empty host), and I think testing the dependency by running conda install in the test section is what's causing problem. I don't think it's a valid approach, likely involving how repodata is made seen by the solver.

**Comment by leofang** [comment]
@conda-forge-admin, please rerender

**Comment by leofang** [comment]
Let's get this in now, and I'll test locally to see if the conflict (https://github.com/conda-forge/cutensor-feedstock/pull/9#issuecomment-780850166) is resolved and report back.

**Comment by leofang** [comment]
(btw, thanks @isuruf for jumping in! Hopefully this will unblock the bot, which currently reports bot error in the 11.1/11.2 migration.)

**Comment by jakirkham** [comment]
Yep thanks for all the work here Leo and Isuru for the assist 😀

**Comment by leofang** [comment]
Seems to work! The same package can work with 11.0/11.2.
```
$ conda create -n test -c conda-forge cudatoolkit=11.0 cutensor
Collecting package metadata (current_repodata.json): done
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.8.3
  latest version: 4.9.2

Please update conda by running

    $ conda update -n base -c defaults conda



## Package Plan ##

  environment location: /home/leofang/miniconda3/envs/test

  added / updated specs:
    - cudatoolkit=11.0
    - cutensor


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    cudatoolkit-11.0.3         |       h15472ef_8       951.9 MB  conda-forge
    cutensor-1.2.2.5           |       h96e36e3_3        88.7 MB  conda-forge
    ------------------------------------------------------------
                                           Total:        1.02 GB

The following NEW packages will be INSTALLED:

  _libgcc_mutex      conda-forge/linux-64::_libgcc_mutex-0.1-conda_forge
  _openmp_mutex      conda-forge/linux-64::_openmp_mutex-4.5-1_gnu
  cudatoolkit        conda-forge/linux-64::cudatoolkit-11.0.3-h15472ef_8
  cutensor           conda-forge/linux-64::cutensor-1.2.2.5-h96e36e3_3
  libgcc-ng          conda-forge/linux-64::libgcc-ng-9.3.0-h2828fa1_18
  libgomp            conda-forge/linux-64::libgomp-9.3.0-h2828fa1_18
  libstdcxx-ng       conda-forge/linux-64::libstdcxx-ng-9.3.0-h6de172a_18


Proceed ([y]/n)? n


CondaSystemExit: Exiting.

$ conda create -n test -c conda-forge cudatoolkit=11.2 cutensor
Collecting package metadata (current_repodata.json): done
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.8.3
  latest version: 4.9.2

Please update conda by running

    $ conda update -n base -c defaults conda



## Package Plan ##

  environment location: /home/leofang/miniconda3/envs/test

  added / updated specs:
    - cudatoolkit=11.2
    - cutensor


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    cudatoolkit-11.2.1         |       h8204236_8       877.7 MB  conda-forge
    cutensor-1.2.2.5           |       h96e36e3_3        88.7 MB  conda-forge
    ------------------------------------------------------------
                                           Total:       966.4 MB

The following NEW packages will be INSTALLED:

  _libgcc_mutex      conda-forge/linux-64::_libgcc_mutex-0.1-conda_forge
  _openmp_mutex      conda-forge/linux-64::_openmp_mutex-4.5-1_gnu
  cudatoolkit        conda-forge/linux-64::cudatoolkit-11.2.1-h8204236_8
  cutensor           conda-forge/linux-64::cutensor-1.2.2.5-h96e36e3_3
  libgcc-ng          conda-forge/linux-64::libgcc-ng-9.3.0-h2828fa1_18
  libgomp            conda-forge/linux-64::libgomp-9.3.0-h2828fa1_18
  libstdcxx-ng       conda-forge/linux-64::libstdcxx-ng-9.3.0-h6de172a_18


Proceed ([y]/n)? n


CondaSystemExit: Exiting.

```

**Pull request merged**